osu.Game/Overlays/OverlayScrollContainer.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using osu.Framework.Allocation;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Effects;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics.Containers;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays
+{
+    public class OverlayScrollContainer : OsuScrollContainer
+    {
+        public ScrollToTopButton Button { get; }
+
+        private bool animaionIsRunning;
```

Comment:
Typo in name (should be at least `animationIsRunning`). Old name was probably better, but I'd ideally want to see this entirely removed.